CodeGuidines.md

```diff
@@ -0,0 +1,138 @@
+This file contains a few general rules on how code should be witten.
+Those guidelines aim to make written code more readable and to have
+an equal style across all files.
+
+<br><br>
+
+# Naming Conventions
+## Repositories
+Repository names should be **snake_case** (all lowercase connected with "`_`")
+
+<br>
+
+## Files
+Just like Repositories, file names should be **snake_case**.
+
+<br>
+
+## Functions
+Function names require to be written in **camelCase**, where the **first letter** is **lowercase**.
+
+
+## Classes
+Class name should be written in **PascalCase**. (*camelCase* with **first letter capitalized**)
+
+## Types
+Similarly to **Classes**, Types should be written in **PascalCase** with an added **_T** at the end.
+
+Example:
+```cpp
+class Vec2_T
+```
```

Comment:
Maybe also add the following sentence:

This also includes structures whose primary use is a simple data container:

```cpp
struct coord_t
{
    int x, y, z;
}
```